### PR TITLE
Client API

### PR DESCRIPTION
**Pull request by Jenil-narola**

### CODE REVIEW

**Comment by smit-luvani** [comment]
- Add response_code
- Fix Indexing
- Remove Comments

**Comment by smit-luvani** [comment]
- Separate Client Plan API [Add, Update, Remove]

**Comment by smit-luvani** [comment]
- Remove Unnecessary async
- Give proper API route name
- Correct response code
- Write proper response messages. Follow [this](https://developers.google.com/webmaster-tools/search-console-api-original/v3/errors)

**Comment by smit-luvani** [comment]
- Date Validation
- Give Proper Variable Name
- Put space after //

File: routes/admin/v1/components/client.js [hunk 1]
```diff
@@ -154,7 +158,7 @@ client.post('/plan/add', (req, res) => {
         return response(res, 400, 'invalid', 'Duration Value invalid', undefined, 'A-4.4.5')
     }
     pushData = {
-        startdate: startDate,
+        startDate: startDate,
```
Comment (smit-luvani) [inline routes/admin/v1/components/client.js:161]: start_date for database

File: routes/admin/v1/components/client.js [hunk 2]
```diff
@@ -182,15 +186,18 @@ client.post('/plan/update', (req, res) => {
...
     }
-    if (req.body.startdate) {
-        var startdate = String(req.body.startdate)
-        pushData.startdate = startdate
+    if (req.body.start_date) {
+        if (String(new Date(req.body.start_date)) == "Invalid Date") {
+            return response(res, 400, 'invalid', 'Invalid Date', undefined, 'A-4.5.5')
+        }
+        var startDate = String(req.body.start_date)
+        pushData.startDate = startDate
```
Comment (smit-luvani) [inline routes/admin/v1/components/client.js:200]: pushData.start_date

Comment (Jenil-narola) [reply routes/admin/v1/components/client.js:200]: ok

File: routes/admin/v1/components/client.js (hunk 1 above)
Comment (Jenil-narola) [reply routes/admin/v1/components/client.js:161]: Done

**Comment by smit-luvani** [comment]
Thanks for Contribution

**Pull request merged**